araproc/framework/data_visualization.py

```diff
@@ -92,7 +95,7 @@ def plot_waveform_bundle(
     
     # make limits look nice
     if time_or_freq == "freq":
-        ax.set_ylim([1,4])
+        ax.set_ylim([1,ymax+0.2])
```

Comment:
Also ylim of `ax.set_ylim(-50, 50)` should catch the whole y axis in `dBm` but you can go with `-50 to ymax+1` or something similar for `ymin` as `ymin-1`